### DISCUSSION TITLE
Integrate with Jetbrains onSave

### DISCUSSION THREAD

**Discussion by whjvenyl** [biomejs/biome#1052] [Ideas]
Provide an onSaveHandler for IntelliJ / Webstorm to run the biome formatting automatically on saving a file.
Implementation similar to the official prettier-plugin:

https://github.com/JetBrains/intellij-plugins/blob/975217c366b0dd429445046bded6213f6d929d39/prettierJS/resources/META-INF/plugin.xml#L47
https://github.com/JetBrains/intellij-plugins/blob/975217c366b0dd429445046bded6213f6d929d39/prettierJS/src/com/intellij/prettierjs/PrettierActionOnSave.java#L18
<img width="1094" alt="CleanShot 2023-12-05 at 01 22 36@2x" src="https://github.com/biomejs/biome/assets/1174049/c72e5ca7-8415-45f5-9040-7e3edb3ab162">

**Comment by victor-teles** [comment]
@whjvenyl I think it's good idea, do you want to work on this?

**Reply by whjvenyl** [reply]
@victor-teles feel free to write it out for anyone to pick up. i'm currently down with quite a bit of work stuff and have yet to get familiar with kotlin again (..mostly doing frontend work atm)

**Comment by victor-teles** [comment]
@whjvenyl This feature is being implemented here https://github.com/biomejs/biome-intellij/pull/3

If you have some feedback or want to see other features in the plugin, Feel free to comment in [☂️ JetBrains plugin 1.0](https://github.com/biomejs/biome/issues/1189)
